modules/apps/dynamic-data-lists/dynamic-data-lists-service/src/main/java/com/liferay/dynamic/data/lists/service/impl/DDLRecordLocalServiceImpl.java

```diff
@@ -892,42 +892,44 @@ public DDLRecord updateStatus(
 		DDLRecord record = ddlRecordPersistence.findByPrimaryKey(
 			recordVersion.getRecordId());
 
-		if (status == WorkflowConstants.STATUS_APPROVED) {
-			int compare = DLUtil.compareVersions(
-				record.getVersion(), recordVersion.getVersion());
-
-			if (compare <= 0) {
-				record.setVersionUserId(recordVersion.getUserId());
-				record.setVersionUserName(recordVersion.getUserName());
-				record.setDDMStorageId(recordVersion.getDDMStorageId());
-				record.setRecordSetId(recordVersion.getRecordSetId());
-				record.setVersion(recordVersion.getVersion());
-				record.setDisplayIndex(recordVersion.getDisplayIndex());
-
-				record = ddlRecordPersistence.update(record);
+		if (status != WorkflowConstants.STATUS_PENDING) {
+			if (status == WorkflowConstants.STATUS_APPROVED) {
+				int compare = DLUtil.compareVersions(
+					record.getVersion(), recordVersion.getVersion());
+
+				if (compare <= 0) {
+					record.setVersionUserId(recordVersion.getUserId());
+					record.setVersionUserName(recordVersion.getUserName());
+					record.setDDMStorageId(recordVersion.getDDMStorageId());
+					record.setRecordSetId(recordVersion.getRecordSetId());
+					record.setVersion(recordVersion.getVersion());
+					record.setDisplayIndex(recordVersion.getDisplayIndex());
+
+					record = ddlRecordPersistence.update(record);
+				}
 			}
-		}
-		else {
-			if (Objects.equals(
-					record.getVersion(), recordVersion.getVersion())) {
+			else {
+				if (Objects.equals(
+						record.getVersion(), recordVersion.getVersion())) {
 
```

Comment:
@michaelmdelcavalcanti can we change to 

```
if (status == WorkflowConstants.STATUS_APPROVED) {
}
else if (status != WorkflowConstants.STATUS_PENDING && 
                Objects.equals(record.getVersion(), recordVersion.getVersion())) {
}
```
?

Reply:
Thanks for your suggestion.